backend/lib/endpoints/posts.js

```diff
@@ -127,19 +216,102 @@ async function routes(app) {
 
   app.patch(
     "/:postId",
-    { preValidation: [app.authenticate], schema: updatePostSchema },
+    {
+      preValidation: [app.authenticate],
+      schema: updatePostSchema,
+    },
     async (req, reply) => {
-      // todo: make sure user can only update their own post
+      // TODO: get userId from jwt
+      // userId = ?
+      // user = User.findById(userId);
+      var user = await User.findById("5ea6900c0e0419d4cb123611");
+      if (user === null) return new httpErrors.Unauthorized();
+
       const post = await Post.findById(req.params.postId);
-      if (post === null) {
-        return reply.send(new httpErrors.NotFound());
-      }
-      Object.keys(req.body).forEach((key) => {
-        if (post[key] && post[key] !== req.body[key]) {
-          post[key] = req.body[key];
+      if (!post) return new httpErrors.NotFound();
+      else if (post.author.authorId != user.id)
+        return new httpErrors.Forbidden();
+      var body = req.body;
+
+      // ExpireAt needs to calculate the date
+      if ("expireAt" in body) {
+        switch (body.expireAt) {
+          case "day":
```

Comment:
same as previous comment with switch case or use object as map
this one can definitely be extracted to a reusable utility function (or maybe we can use something from another date library directly)  Like @joaofnds said probably Moment or date-fn's are widely used and have a lot of features, I've also seen https://moment.github.io/luxon/ which is a smaller and newer alternative to moment... so depends a bit what we need